src/utils/asyncStoreHelper.js

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// @flow
+
+import { asyncStorage } from "devtools-modules";
+
+/*
+ * asyncStoreHelper wraps asyncStorage so that it is easy to define project
+ * specific properties. It is similar to PrefsHelper.
+ *
+ * e.g.
+ *   const asyncStore = asyncStoreHelper("r", {a: "_a"})
+ *   asyncStore.a         // => asyncStorage.getItem("r._a")
+ *   asyncStore.a = 2     // => asyncStorage.setItem("r._a", 2)
+ */
+export function asyncStoreHelper(root: string, mappings: Object) {
+  let store: any = {};
+  Object.keys(mappings).map(key =>
+    Object.defineProperty(store, key, {
+      get() {
+        return asyncStorage.getItem(`${root}.${mappings[key]}`);
+      },
+      set(value) {
+        return asyncStorage.setItem(`${root}.${mappings[key]}`, value);
+      }
+    })
+  );
+
+  return (store: { [$Keys<typeof mappings>]: any });
```

Comment:
this is pretty cool...

if you try and do `asyncStore.foo` it will fail unless `foo` is in a key in the `mappings` object!
I tried this with `asyncStore.pendingBreakpoints2` and got a flow error

Reply:
awesome